synapse/storage/databases/main/devices.py

```diff
@@ -1121,18 +1189,27 @@ async def store_device(
             raise StoreError(500, "Problem storing device.")
 
     async def delete_device(self, user_id: str, device_id: str) -> None:
-        """Delete a device.
+        """Delete a device and its device_inbox.
 
         Args:
             user_id: The ID of the user which owns the device
             device_id: The ID of the device to delete
         """
-        await self.db_pool.simple_delete_one(
-            table="devices",
-            keyvalues={"user_id": user_id, "device_id": device_id, "hidden": False},
-            desc="delete_device",
-        )
 
+        def _delete_device_txn(txn: LoggingTransaction) -> None:
+            self.db_pool.simple_delete_one_txn(
+                txn,
+                table="devices",
+                keyvalues={"user_id": user_id, "device_id": device_id, "hidden": False},
+            )
+
+            self.db_pool.simple_delete_txn(
+                txn,
+                table="device_inbox",
+                keyvalues={"user_id": user_id, "device_id": device_id},
+            )
+
+        await self.db_pool.runInteraction("delete_device", _delete_device_txn)
         self.device_id_exists_cache.invalidate((user_id, device_id))
```

Comment:
I wonder if this entire method should just be:

```python
await self.delete_devices(user_id, [device_id])
```

Instead of duplicating the logic of how to delete these two tables.